lib/console_yasttest.pm

```diff
@@ -17,9 +17,9 @@ sub post_fail_hook {
 
     show_tasks_in_blocked_state if ($defer_blocked_task_info);
 
-    upload_logs('/var/log/zypper.log');
     $self->remount_tmp_if_ro;
     $self->save_upload_y2logs;
+    upload_logs('/var/log/zypper.log');
```

Comment:
why you need to change order ? I suspecting that you trying to make sure that the most important logs will be uploaded no matter if ones which are less important will fail. If this is the case maybe you just need to use `fail_ok => 1` option available for `upload_logs` function ?

Reply:
Actually, we just do a lot of useful things in save_upload_y2logs, one of which is network recovery. In case we will call upload_logs without network, we will fail and won't even try to collect further logs. By reordering we will attempt to fix network before uploading anything.